app/src/main/java/com/example/lunchmunch/ShoppingListActivity.java

```diff
@@ -356,11 +291,23 @@ public void onSuccess(Object o) {
                         Optional<Ingredient> optExistingIngr = shoppingList.stream().
                                 filter(i -> i.getName().equals(ingredient.getName())).
                                 findFirst(); // can use findFirst as we know that ingr name is its unique id so will have only one occurance
-                        Ingredient existingIngr = optExistingIngr.orElse(null);
-                        if (existingIngr != null) {
+                        Ingredient shopIngr = optExistingIngr.orElse(null);
+                        if (shopIngr != null) {
                             // if the amount of ingr we required in the shopping list is still greater then the amount the user entered then just update the ingr in the shopping list
                             // since we create a new Ingr instance in addShopIngrFragment we cant directly acess we have to access by name to delete
-                            shoppingList.remove(existingIngr);
+
+                            Float diff = shopIngr.getCount() - ingredient.getCount();
+
+                            System.out.println();
+
+                            // if the count of the ingr the user entered is less than or equal to the amount required
+                            if (diff > 0.01) {
```

Comment:
Just curious as to why it's 0.01  and not just 0

Reply:
Basically doing 
if (shopIngr.getCount() > ingredient.getCount())
but since we are using floats there is some odd arithmetic behaviour this makes sure that one is significantly bigger than the other  (significantly meaning 10% hence 0.01)